src/signals/incident/components/form/MapSelectors/Asset/Selector/Nearby/Nearby.tsx

```diff
@@ -0,0 +1,187 @@
+import { useFetch } from 'hooks'
+import { useCallback, useContext, useEffect, useRef, useState } from 'react'
+import { useSelector } from 'react-redux'
+import L from 'leaflet'
+import { useMapInstance } from '@amsterdam/react-maps'
+
+import type { FC } from 'react'
+import type { FeatureCollection, Feature } from 'geojson'
+import type { LatLngTuple, LeafletMouseEvent } from 'leaflet'
+import type { Location } from 'types/incident'
+
+import reverseGeocoderService from 'shared/services/reverse-geocoder'
+import { makeSelectCategory } from 'signals/incident/containers/IncidentContainer/selectors'
+import configuration from 'shared/services/configuration/configuration'
+import { featureToCoordinates } from 'shared/services/map-location'
+
+import { formatDate } from 'signals/incident/containers/IncidentReplyContainer/utils'
+import useBoundingBox from 'signals/incident/components/form/MapSelectors/hooks/useBoundingBox'
+import type { Item } from 'signals/incident/components/form/MapSelectors/types'
+import { NEARBY_TYPE } from 'signals/incident/components/form/MapSelectors/constants'
+import AssetSelectContext from 'signals/incident/components/form/MapSelectors/Asset/context'
+
+// Custom Point type, because the compiler complains about the coordinates type
+type Point = {
+  type: 'Point'
+  coordinates: LatLngTuple
+}
+
+type Properties = {
+  category: {
+    name: string
+  }
+  created_at: string
+}
+
+type NearbyMarker = L.Marker<Properties>
+
+interface MarkerMouseEvent extends LeafletMouseEvent {
+  sourceTarget: NearbyMarker
+}
+
+export const nearbyMarkerIcon = L.icon({
+  iconSize: [24, 32],
+  iconUrl: '/assets/images/area-map/icon-pin.svg',
+})
+
+export const nearbyMarkerSelectedIcon = L.icon({
+  iconSize: [40, 40],
+  iconUrl: '/assets/images/area-map/icon-pin-red.svg',
+})
+
+const formattedDate = (date: string) =>
+  formatDate(new Date(date), `'Gemeld op:' dd MMMM`)
+
+const onFeatureGroupClick = /* istanbul ignore next */ ({
+  sourceTarget, // clicked marker
+  target, // feature group
+}: LeafletMouseEvent) => {
+  const targetElement = sourceTarget.getElement()
+
+  target.eachLayer((layer: L.Marker<Properties>) => {
+    const element = layer.getElement()
+
+    if (!element) return
+
+    if (element !== targetElement) {
+      layer.setIcon(nearbyMarkerIcon)
+    }
+  })
+}
+
+/**
+ * Component using the Leaflet API to plot markers in a feature layer on a map
+ * Because the markers are rendered in a feature group, events can be captured by that group.
+ * Which is useful, because the all of the layers in the feature group can be easily targeted.
```

Comment:
What is the purpose of the featureGroupClick? Can you specify an example where all of the layers need to be targeted at once?